File: frontend/src/api/monitoring.ts
import { fetchWithAuth } from "./client";

export type TimeSeriesPoint = {
  time: string;
  cpu: number;
  memory: number;
  latency: number;
};

export type SystemHealthResponse = {
  status: string;
  uptime: string;
  cpu_usage: number;
  memory_usage: number;
  api_latency_ms: number;
  prediction_volume: number;
  history: TimeSeriesPoint[];
};

export async function getSystemHealth(): Promise<SystemHealthResponse> {
  const res = await fetchWithAuth(`/monitoring/health`);
  if (!res.ok) {
    const error = await res.json();
    throw new Error(error.detail || "Failed to fetch health data");
  }
  return res.json();
}

File: backend/app/schemas/monitoring.py
from pydantic import BaseModel
from typing import List

class TimeSeriesPoint(BaseModel):
    time: str
    cpu: float
    memory: float
    latency: float

class SystemHealthResponse(BaseModel):
    status: str
    uptime: str
    cpu_usage: float
    memory_usage: float
    api_latency_ms: float
    prediction_volume: int
    history: List[TimeSeriesPoint]
